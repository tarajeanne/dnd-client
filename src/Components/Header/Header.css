*{
  box-sizing: border-box;
}

.Header {
  height: 120px;
  max-width: 1050px;
  padding: 0 15px;
  margin: 50px auto 0;
  background-image: url("https://ih0.redbubble.net/image.214383515.3210/flat,800x800,075,f.jpg");
  background-position: 97% 100%;
  background-repeat: no-repeat;
  background-size: contain;

}

.hero {
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
 
}

h1{
  font-family: 'Raleway', sans-serif;
  margin-bottom: 0px; 
}

.sub {
  margin-top: 5px;
  font-family:'Montserrat', sans-serif;
}

.header-image {
  height: 115px;
  float: right;
}

.user-controls {
  position: absolute;
  top: 0;
  width: 100%;
  background-color: #7A6263;
  color: white;
  height: 40px;
  text-align: right;
  font-family:'Montserrat', sans-serif;
  font-size: 16px;
  font-weight: bold;
}

.user-links {
  margin: auto;
  padding: 0 15px;
  padding-top: 10px;
  list-style: none;
  max-width: 1050px;
}

.user-link {
  display: inline;
  margin-right: 0 5px;
}

.user-link:not(:first-child):before{
  content: '    |    ';
}

li > a:link, li > a:visited {
  color: white;
}

li > a:hover {
  color: #C6DEA6;
}