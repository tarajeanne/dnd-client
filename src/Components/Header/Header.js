import React from 'react';
import PropTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import TokenService from '../../services/token-service';
import IdleService from '../../services/idle-service';
import './Header.css';

class Header extends React.Component {
  static propTypes = {
    hasAuthToken: PropTypes.bool
  };

  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
  };

  renderUserLinks = () => {
    return (
      <ul className="user-links">
        <li className="user-link">
          <NavLink to="/user" activeClassName="active-link">
            Home
          </NavLink>
        </li>
        <li className="user-link">
          <Link onClick={this.handleLogoutClick} to="/">
            Logout
          </Link>
        </li>
      </ul>
    );
  };

  render() {
    return (
      <div className="Header">
        <div className="header-container">
          <header className="hero">
            <h1>DnD Character Creator</h1>
            <p className="sub">for Absolute Beginners</p>
          </header>
          <nav className="user-controls">
            {this.props.hasAuthToken ? this.renderUserLinks() : ''}
          </nav>
        </div>
      </div>
    );
  }
}

export default Header;
