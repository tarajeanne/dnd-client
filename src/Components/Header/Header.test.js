import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Header', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
