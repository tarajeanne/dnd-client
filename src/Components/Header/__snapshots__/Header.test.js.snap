// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected 1`] = `
<Fragment>
  <div
    className="Header"
  >
    <header
      className="hero"
    >
      <h1>
        DnD Character Creator
      </h1>
      <p
        className="sub"
      >
        for Absolute Beginners
      </p>
    </header>
    <nav
      className="user-controls"
    >
      <ul
        className="user-links"
      >
        <li
          className="user-link"
        >
          <NavLink
            activeClassName="active-link"
            to="/login"
          >
            Login
          </NavLink>
        </li>
        <li
          className="user-link"
        >
          <NavLink
            activeClassName="active-link"
            to="/register"
          >
            Register
          </NavLink>
        </li>
      </ul>
    </nav>
  </div>
</Fragment>
`;
